### PR TITLE
Replace heap_getnext

### PR DESCRIPTION
**Pull request by markgomer**
The changes made in this PR pass the installcheck tests on AGE's master branch. 

For now, this is only working while replacing on agtype.c file.

### CODE REVIEW

**Comment by rafsun42** [comment]
@markgomer Could you send this PR to this branch instead- https://github.com/rafsun42/age/tree/table-replace-heap

File: src/backend/utils/adt/agtype.c [hunk 1]
```diff
@@ -4872,7 +4873,11 @@ static Datum get_vertex(const char *graph, const char *vertex_label,
     /* open the relation (table), begin the scan, and get the tuple  */
     graph_vertex_label = table_open(vertex_label_table_oid, ShareLock);
     scan_desc = table_beginscan(graph_vertex_label, snapshot, 1, scan_keys);
-    tuple = heap_getnext(scan_desc, ForwardScanDirection);
+    //tuple = heap_getnext(scan_desc, ForwardScanDirection);
```
Comment (rafsun42) [inline src/backend/utils/adt/agtype.c]: It is okay to remove this comment.

**Comment by markgomer** [comment]
> @markgomer Could you send this PR to this branch instead- https://github.com/rafsun42/age/tree/table-replace-heap

Changed it! I made a force push to keep the master branch's changes away from this one.

File: src/backend/utils/adt/agtype.c (hunk 1 above)
Comment (markgomer) [reply src/backend/utils/adt/agtype.c]: Done!

**Comment by markgomer** [comment]
Replaced all 9 instances of heap_getnext in AGE:

- [x] age/src/backend/catalog/ag_label.c:
  314  
  315:         tuple = heap_getnext(scan_desc, ForwardScanDirection);
  316  

- [x] age/src/backend/executor/cypher_delete.c:
  429          /* Retrieve the tuple. */
  430:         heap_tuple = heap_getnext(scan_desc, ForwardScanDirection);
  431  

  513  
  514:         tuple = heap_getnext(scan_desc, ForwardScanDirection);
  515  

- [x] age/src/backend/executor/cypher_set.c:
  585              /* Retrieve the tuple. */
  586:             heap_tuple = heap_getnext(scan_desc, ForwardScanDirection);
  587  

- [x] age/src/backend/executor/cypher_utils.c:
  194  
  195:     tuple = heap_getnext(scan_desc, ForwardScanDirection);
  196  

- [x] age/src/backend/utils/adt/age_global_graph.c:
  224      /* get all of the label names */
  225:     while((tuple = heap_getnext(scan_desc, ForwardScanDirection)) != NULL)
  226      {

  436          /* get all tuples in table and insert them into graph hashtables */
  437:         while((tuple = heap_getnext(scan_desc, ForwardScanDirection)) != NULL)
  438          {

  535          /* get all tuples in table and insert them into graph hashtables */
  536:         while((tuple = heap_getnext(scan_desc, ForwardScanDirection)) != NULL)
  537          {

- [x] age/src/backend/utils/adt/agtype.c:
  4779      scan_desc = table_beginscan(graph_vertex_label, snapshot, 1, scan_keys);
  4780:     tuple = heap_getnext(scan_desc, ForwardScanDirection);
  4781

**Comment by rafsun42** [comment]
@markgomer 

The PR compiled, but threw a warning.
```
src/backend/utils/adt/agtype.c: In function ‘get_vertex’:
src/backend/utils/adt/agtype.c:4870:21: warning: unused variable ‘slot’ [-Wunused-variable]
 4870 |     TupleTableSlot* slot;
      | 
```

**Comment by markgomer** [comment]
@rafsun42, just removed it

**Comment by rafsun42** [comment]
@markgomer I am also wondering, does a `HeapTuple` survive in the memory after `ExecDropSingleTupleTableSlot` is called? My concern is we are returning the HeapTuple after its slot is dropped.

**Comment by markgomer** [comment]
> [@markgomer](https://github.com/markgomer) I am also wondering, does a `HeapTuple` survive in the memory after `ExecDropSingleTupleTableSlot` is called? My concern is we are returning the HeapTuple after its slot is dropped.

@rafsun, I did some tests now and the `HeapTuple` did survive in memory after `ExecDropSingleTupleTableSlot` was called. Its contents were still accessible.

There is also a test after the "drop" is done, in line 4892:
`if (!HeapTupleIsValid(tuple))`, and it went ok in all tests.

From what I saw, the function `ExecDropSingleTupleTableSlot` only releases the resources associated with the `TupleTableSlot`, not the `HeapTuple` itself, so it should be safe to make it that way, if I'm not mistaken, of course.

**Pull request closed without merging**